webviz_subsurface_components/py_expression_eval.py

```diff
@@ -0,0 +1,483 @@
+        if (noperators + 1) != len(tokenstack):
+            self.error_parsing(self.pos, "parity")
+
+        return Expression(tokenstack, self.ops1, self.ops2)
+
+    def evaluate(self, expr, variables):
+        return self.parse(expr).evaluate(variables)
+
+    def error_parsing(self, column, msg):
+        self.success = False
+        self.errormsg = (
+            "parse error [column "
+            + str(column)
+            + "]: "
+            + msg
+            + ", expression: "
+            + self.expression
+        )
+        raise Exception(self.errormsg)
+
+    def addfunc(self, tokenstack, operstack, type_):
+        operator = Token(
+            type_,
+            self.tokenindex,
+            self.tokenprio + self.tmpprio,
+            0,
+        )
+        while len(operstack) > 0:
+            if operator.prio_ <= operstack[len(operstack) - 1].prio_:
+                tokenstack.append(operstack.pop())
+            else:
+                break
+        operstack.append(operator)
+
+    def isNumber(self):
+        r = False
+
+        if self.expression[self.pos] == "E":
+            return False
+
+        # number in scientific notation
+        pattern = r"([-+]?([0-9]*\.?[0-9]*)[eE][-+]?[0-9]+).*"
+        match = re.match(pattern, self.expression[self.pos :])
+        if match:
+            self.pos += len(match.group(1))
+            self.tokennumber = float(match.group(1))
+            return True
+
+        # number in decimal
+        str = ""
+        while self.pos < len(self.expression):
+            code = self.expression[self.pos]
+            if (code >= "0" and code <= "9") or code == ".":
+                if len(str) == 0 and code == ".":
+                    str = "0"
+                str += code
+                self.pos += 1
+                try:
+                    self.tokennumber = int(str)
+                except ValueError:
+                    self.tokennumber = float(str)
+                r = True
+            else:
+                break
+        return r
+
+    def isConst(self):
+        for i in self.consts:
+            L = len(i)
+            str = self.expression[self.pos : self.pos + L]
+            if i == str:
+                if len(self.expression) <= self.pos + L:
+                    self.tokennumber = self.consts[i]
+                    self.pos += L
+                    return True
+                if (
+                    not self.expression[self.pos + L].isalnum()
+                    and self.expression[self.pos + L] != "_"
+                ):
+                    self.tokennumber = self.consts[i]
+                    self.pos += L
+                    return True
+        return False
+
+    def isOperator(self):
+        ops = (
+            ("**", 8, "**"),
+            ("^", 8, "^"),
+            ("%", 6, "%"),
+            ("/", 6, "/"),
+            (u"\u2219", 5, "*"),  # bullet operator
+            (u"\u2022", 5, "*"),  # black small circle
```

Comment:
```suggestion
            ("\u2219", 5, "*"),  # bullet operator
            ("\u2022", 5, "*"),  # black small circle
```
Unicode by default on Python 3.